pkg/reportingplugins/mercury/types.go

```diff
@@ -0,0 +1,53 @@
+package mercury
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+// todo: group items by function in separate interfaces
+type ParsedObservation interface {
+	GetTimestamp() uint32
+	GetObserver() commontypes.OracleID
+	GetBenchmarkPrice() *big.Int
+	GetBid() *big.Int
+	GetAsk() *big.Int
+	GetPricesValid() bool
+	GetCurrentBlockNum() int64
+	GetCurrentBlockHash() []byte
+	GetCurrentBlockTimestamp() uint64
+	GetCurrentBlockValid() bool
+	GetMaxFinalizedBlockNumber() int64
+	GetMaxFinalizedBlockNumberValid() bool
+	GetMaxFinalizedTimestamp() int64
```

Comment:
As stated above, an alternative implementation might look like:

```
GetCurrentBlock() (num int64, valid bool)
```

Could reduce the number of methods and make it a bit more readable this way. It also makes it possible to change the underlying schema later.